Prometheus Prefect Exporter doesn't support basic Authentication when implemented on prefect-server

When deploying an OSS Prefect server instance with basic authentication enabled (by setting the environment variable `PREFECT_SERVER_API_AUTH_STRING`) as per [this issue](https://github.com/PrefectHQ/prefect/issues/2238) the Prometheus exporter fails its calls to the prefect-api: 
```
2025-01-17 11:02:24,853 - prometheus-prefect-exporter - [ERROR] 401 Client Error: Unauthorized for url: http://prefect-server.prefect-server.svc.cluster.local:4200/api/deployments/filter
2025-01-17 11:02:24,866 - prometheus-prefect-exporter - [ERROR] 401 Client Error: Unauthorized for url: http://prefect-server.prefect-server.svc.cluster.local:4200/api/deployments/filter
2025-01-17 11:02:24,876 - prometheus-prefect-exporter - [ERROR] 401 Client Error: Unauthorized for url: http://prefect-server.prefect-server.svc.cluster.local:4200/api/deployments/filter
401 Client Error: Unauthorized for url: http://prefect-server.prefect-server.svc.cluster.local:4200/api/deployments/filter
```

I note there is an environment variable available (`PREFECT_API_KEY`) for the exporter to configure auth, however this sets a 'Bearer' token ([here](https://github.com/PrefectHQ/prometheus-prefect-exporter/blob/main/main.py#L36)) rather than allowing for 'Basic' auth).

--- a/README.md
+++ b/README.md
@@ -92,6 +92,7 @@ You can now reach each service locally:
 You can modify environment variables to change the behavior of the exporter.
 - An API Key is only required for auth-enabled, on-prem, self-managed solutions.
 - An API key is not required for open-source or Prefect Server.
+- If an API key and API auth string are provided, then the API key takes precedence.
 
 | Environment Variable | Description | Default |
 | --- | --- | --- |
@@ -101,7 +102,8 @@ You can modify environment variables to change the behavior of the exporter.
 | `METRICS_PORT` | Port to expose metrics on | `8000` |
 | `OFFSET_MINUTES` | Number of minutes to offset the start time when fetching metrics from Prefect API | `5` |
 | `PREFECT_API_URL` | Prefect API URL | `https://localhost:4200/api` |
-| `PREFECT_API_KEY` | Prefect API KEY (Optional) | `""` |
+| `PREFECT_API_KEY` | Prefect API key (Optional) | `""` |
+| `PREFECT_API_AUTH_STRING` | Prefect API auth string, automatically base64-encoded (Optional) | `""` |
 | `PREFECT_CSRF_ENABLED` | Enable compatibilty with Prefect Servers using CSRF protection | `False` |
 | `PAGINATION_ENABLED` | Enable pagination usage. (Uses more resources) | `True` |
 | `PAGINATION_LIMIT` | Pagination limit | `200` |

--- a/compose.yml
+++ b/compose.yml
@@ -5,6 +5,7 @@ services:
       - "4200:4200"
     environment:
       PREFECT_LOGGING_LEVEL: debug
+      # PREFECT_SERVER_API_AUTH_STRING: ""
     command:
       - prefect
       - server
@@ -23,6 +24,7 @@ services:
       - "8000:8000"
     environment:
       PREFECT_API_URL: http://prefect:4200/api
+      # PREFECT_API_AUTH_STRING: ""
     depends_on:
       prefect:
         condition: service_healthy

--- a/main.py
+++ b/main.py
@@ -1,4 +1,5 @@
 import os
+import base64
 import logging
 import time
 import uuid
@@ -21,6 +22,7 @@ def metrics():
     offset_minutes = int(os.getenv("OFFSET_MINUTES", "3"))
     url = str(os.getenv("PREFECT_API_URL", "http://localhost:4200/api"))
     api_key = str(os.getenv("PREFECT_API_KEY", ""))
+    api_auth_string = str(os.getenv("PREFECT_API_AUTH_STRING", ""))
     csrf_client_id = str(uuid.uuid4())
     scrape_interval_seconds = int(os.getenv("SCRAPE_INTERVAL_SECONDS", "30"))
     # Configure logging
@@ -32,8 +34,14 @@ def metrics():
     # Configure headers for HTTP requests
     headers = {"accept": "application/json", "Content-Type": "application/json"}
 
+    if api_auth_string:
+        api_auth_string_encoded = base64.b64encode(api_auth_string.encode("utf-8")).decode("utf-8")
+        headers["Authorization"] = f"Basic {api_auth_string_encoded}"
+        logger.info("Added Basic Authorization header for PREFECT_API_AUTH_STRING")
+
     if api_key:
         headers["Authorization"] = f"Bearer {api_key}"
+        logger.info("Added Bearer Authorization header for PREFECT_API_KEY")
 
     # check endpoint
     PrefectHealthz(

